nikola/post.py

```diff
@@ -277,12 +277,10 @@ def __repr__(self):
         return '<Post: {0!r} {1}>'.format(self.source_path, m.hexdigest())
 
     def _has_pretty_url(self, lang):
-        if self.pretty_urls and \
-                self.meta[lang].get('pretty_url', '') != 'False' and \
-                self.meta[lang]['slug'] != 'index':
-            return True
-        else:
-            return False
+        m = self.meta[lang].get('pretty_url', '')
+        return ((self.pretty_urls and m != 'False' and
+                 self.meta[lang]['slug'] != 'index') or
```

Comment:
Shouldn't `self.meta[lang]['slug'] != 'index'` be `and`ed with both sides of `or`? I.e. something like
```
    return self.meta[lang]['slug'] != 'index' and ((self.pretty_urls and m != 'False') or (not self.pretty_urls and m == 'True'))
```